modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v3_9_1/DDMStructureUpgradeProcess.java

```diff
@@ -86,17 +86,20 @@ private void _upgradeDDMFormFieldReference(DDMFormField ddmFormField) {
 	private void _upgradeDDMStructure() throws Exception {
 		try (PreparedStatement preparedStatement1 = connection.prepareStatement(
 				StringBundler.concat(
-					"select DDMStructure.structureId, ",
-					"DDMStructureVersion.definition from DDMStructure inner ",
-					"join DDMStructureVersion on DDMStructure.structureId = ",
-					"DDMStructureVersion.structureId where ",
+					"select DDMStructure.ctCollectionId, ",
+					"DDMStructure.structureId, DDMStructureVersion.definition ",
+					"from DDMStructure inner join DDMStructureVersion on ",
+					"DDMStructure.ctCollectionId = ",
+					"DDMStructureVersion.ctCollectionId AND ",
```

Comment:
I don't think we need this: `DDMStructure.ctCollectionId = DDMStructureVersion.ctCollectionId`
Can you remove and test it, please?

Reply:
On it!

Reply:
Hi @marcelabc 

After testing I noticed that by removing this part of the code both tests `DDMStructureCTUpgradeProcessTest` and `DDMStructureVersionDDMStructureCTUpgradeProcessTest` are failing; therefore I opted to leave the code as it is.